frontend/src/components/ShootWorkers/WorkerInputGeneric.vue

```diff
@@ -362,13 +396,20 @@ export default {
       this.validateInput()
     },
     onUpdateMachineType () {
+      this.setVolumeDependingOnMachineType()
       this.validateInput()
     },
     onUpdateVolumeType () {
       this.validateInput()
     },
     onInputVolumeSize () {
-      this.$v.worker.volume.size.$touch()
+      const machineType = this.selectedMachineType
+      if (!this.canDefineVolumeSize || get(machineType, 'storage.size') === this.volumeSizeInternal) {
+        delete this.worker.volume
```

Comment:
don't understand

Reply:
I added some comments to make it clearer